addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/automation/ui/ScriptJobDialog.java

```diff
@@ -56,42 +73,119 @@ public ScriptJobDialog(ScriptJob job) {
+        return new File(System.getProperty("user.home"));
+    }
+
+    private ScriptAction getScriptAction() {
+        return ScriptJob.createScriptAction(
+                new ScriptJobParameters(this.getStringValue(SCRIPT_ACTION_PARAM)), null);
+    }
+
     private void onScriptActionChanged() {
-        String action = this.getStringValue(SCRIPT_ACTION_PARAM);
-        List<String> scriptTypes = ScriptJob.validScriptTypesForAction(action);
+        ScriptAction sa = getScriptAction();
+        List<String> scriptTypes = sa.getSupportedScriptTypes();
         this.setComboFields(
                 SCRIPT_TYPE_PARAM, scriptTypes, this.job.getData().getParameters().getType());
+
+        List<String> disabledFields = sa.getDisabledFields();
+        for (String fieldName : ALL_FIELDS) {
+            if (disabledFields.contains(fieldName)) {
+                this.getField(fieldName).setEnabled(false);
+                this.setFieldValue(fieldName, "");
+            } else {
+                this.getField(fieldName).setEnabled(true);
+            }
+        }
     }
 
     private void onScriptTypeChanged() {
         String scriptType = this.getStringValue(SCRIPT_TYPE_PARAM);
         List<String> scripts = ScriptAction.getAvailableScriptNames(scriptType);
+        // Always have a blank option at the start
+        scripts.add(0, "");
         this.setComboFields(
                 SCRIPT_NAME_PARAM, scripts, this.job.getData().getParameters().getName());
     }
 
+    private void onScriptNameChanged() {
+        ScriptAction sa = getScriptAction();
+        if (sa.getDisabledFields().contains(SCRIPT_FILE_PARAM)) {
+            String scriptName = this.getStringValue(SCRIPT_NAME_PARAM);
+            ScriptWrapper sw = ScriptAction.getExtScript().getScript(scriptName);
+            if (sw != null) {
+                this.setFieldValue(SCRIPT_FILE_PARAM, sw.getFile().getAbsolutePath());
+            }
+        }
+    }
+
     @Override
     public void save() {
         this.job.getData().setName(this.getStringValue(NAME_PARAM));
         this.job.getData().getParameters().setAction(this.getStringValue(SCRIPT_ACTION_PARAM));
         this.job.getData().getParameters().setType(this.getStringValue(SCRIPT_TYPE_PARAM));
+        this.job.getData().getParameters().setEngine(this.getStringValue(SCRIPT_ENGINE_PARAM));
         this.job.getData().getParameters().setName(this.getStringValue(SCRIPT_NAME_PARAM));
+
+        ScriptAction sa = getScriptAction();
+        if (sa.getDisabledFields().contains(SCRIPT_FILE_PARAM)) {
+            this.job.getData().getParameters().setFile(null);
+        } else {
+            File f = new File(this.getStringValue(SCRIPT_FILE_PARAM));
+            if (f.isFile()) {
+                this.job.getData().getParameters().setFile(this.getStringValue(SCRIPT_FILE_PARAM));
+            }
+        }
         this.job.resetAndSetChanged();
     }
 
     @Override
     public String validateFields() {
```

Comment:
- Action `run` and `remove` should require a script name.
- Action `add` with no engine selected and no engine found from the extension leads to an exception:
  ```
  ERROR org.zaproxy.zap.ZAP.UncaughtExceptionLogger - Exception in thread "AWT-EventQueue-0"
  java.security.InvalidParameterException: No such engine: null
  	at org.zaproxy.zap.extension.script.ExtensionScript.getEngineWrapper(ExtensionScript.java:473) ~[main/:?]
  	at org.zaproxy.zap.extension.scripts.automation.actions.AddScriptAction.getEngineWrapper(AddScriptAction.java:145) ~[?:?]
  	at org.zaproxy.zap.extension.scripts.automation.actions.AddScriptAction.verifyParameters(AddScriptAction.java:112) ~[?:?]
  	at org.zaproxy.zap.extension.scripts.automation.ui.ScriptJobDialog.validateFields(ScriptJobDialog.java:184) ~[?:?]
  ```
- It's possible to add a script with no name.